proj-sys/build.rs

```diff
@@ -3,13 +3,67 @@ use cmake;
 use flate2::read::GzDecoder;
 use std::fs::File;
 
+#[cfg(target_env = "msvc")]
+use vcpkg;
+#[cfg(target_env = "msvc")]
+use build;
 use pkg_config;
 use std::env;
 use std::path::PathBuf;
 use tar::Archive;
 
 const MINIMUM_PROJ_VERSION: &str = "7.2.1";
 
+#[cfg(target_env = "msvc")]
+fn try_vcpkg() -> Result<std::path::PathBuf, Box<dyn std::error::Error>> {
+    build::link("shell32", true);
+    build::link("ole32", true);
+    let lib = vcpkg::Config::new()
+    .emit_includes(true)
+    .find_package("proj");
+
+    if let Err(_e) = lib {
+        eprintln!("vcpkg proj library not found, trying pkg_config");
+        return try_pkg_config()
```

Comment:
To make sure I understand: 

If the target is msvc, we'll prefer vcpkg, but if `vcpkg` fails to find proj, it cannot be found, we'll fall back to pkg_config, and then if that also fails we'll fall back to building from source.

Is that right?

Assuming I've got that right, please slightly rewrite this to avoid the `lib.unwrap` below. Something like:

match lib {
    Ok(include_path) => Ok(include[0]),
    Err(e) => {
        // can we log `e`?
        return try_pkg_config()
    }
}
```

Reply:
Yes your right it's vcpkg -> pkg_config -> build from source
I will look